#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Fix a permission denied during make check
# DP: Author: Daniel Jacobowitz <dan@debian.org>
# DP: Upstream status: Submitted
# DP: Date: 2003-10-12

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
2003-10-12  Daniel Jacobowitz  <drow@mvista.com>

	* Makefile (tests): Make sure check-c++-types.sh is executable.

Index: glibc/Makefile
===================================================================
RCS file: /big/fsf/rsync/glibc-cvs/libc/Makefile,v
retrieving revision 1.234
diff -u -p -r1.234 Makefile
--- glibc/Makefile	9 Sep 2003 02:30:15 -0000	1.234
+++ glibc/Makefile	12 Oct 2003 23:24:08 -0000
@@ -232,6 +232,7 @@ tests: $(objpfx)c++-types-check.out
 ifneq ($(CXX),no)
 ifneq (,$(wildcard scripts/data/c++-types-$(base-machine)-$(config-os).data))
 $(objpfx)c++-types-check.out: scripts/data/c++-types-$(base-machine)-$(config-os).data
+	chmod a+x scripts/check-c++-types.sh
 	scripts/check-c++-types.sh $^ $(CXX) $(filter-out -std=gnu99,$(CFLAGS)) $(CPPFLAGS) > $@
 else
 $(objpfx)c++-types-check.out:
